File: shared-pref-sample/src/main/java/ru/surfstudio/android/shared/pref/sample/ui/base/configurator/ActivityScreenConfigurator.java
package ru.surfstudio.android.shared.pref.sample.ui.base.configurator;

import android.content.Intent;

import ru.surfstudio.android.core.mvp.configurator.BaseActivityViewConfigurator;
import ru.surfstudio.android.sample.dagger.ui.base.dagger.activity.DefaultActivityModule;
import ru.surfstudio.android.sample.dagger.ui.base.dagger.screen.DefaultActivityScreenModule;
import ru.surfstudio.android.shared.pref.sample.ui.base.dagger.activity.ActivityComponent;
import ru.surfstudio.android.shared.pref.sample.ui.base.dagger.activity.DaggerActivityComponent;
import ru.surfstudio.android.shared.pref.sample.app.App;
import ru.surfstudio.android.shared.pref.sample.app.dagger.AppComponent;

/**
 * Базовый конфигуратор для экрана, основанного на активити
 */

public abstract class ActivityScreenConfigurator
        extends BaseActivityViewConfigurator<AppComponent, ActivityComponent, DefaultActivityScreenModule> {

    public ActivityScreenConfigurator(Intent intent) {
        super(intent);
    }

    @Override
    protected ActivityComponent createActivityComponent(AppComponent parentComponent) {
        return DaggerActivityComponent.builder()
                .appComponent(parentComponent)
                .defaultActivityModule(new DefaultActivityModule(getPersistentScope()))
                .build();
    }

    @Override
    protected AppComponent getParentComponent() {
        return ((App) (getTargetActivity()).getApplicationContext()).getAppComponent();
    }

    @Override
    protected DefaultActivityScreenModule getActivityScreenModule() {
        return new DefaultActivityScreenModule(getPersistentScope());
    }
}

File: navigation/sample-standard/src/main/java/ru/surfstudio/android/navigation/sample_standard/di/ui/configurator/ActivityScreenConfigurator.kt
package ru.surfstudio.android.navigation.sample_standard.di.ui.configurator

import android.content.Intent
import ru.surfstudio.android.core.mvp.configurator.BaseActivityViewConfigurator
import ru.surfstudio.android.navigation.sample_standard.App
import ru.surfstudio.android.navigation.sample_standard.di.AppComponent
import ru.surfstudio.android.navigation.sample_standard.di.ui.ActivityComponent
import ru.surfstudio.android.navigation.sample_standard.di.ui.ActivityNavigationModule
import ru.surfstudio.android.navigation.sample_standard.di.ui.DaggerActivityComponent
import ru.surfstudio.android.sample.dagger.ui.base.dagger.activity.DefaultActivityModule
import ru.surfstudio.android.sample.dagger.ui.base.dagger.screen.DefaultActivityScreenModule

abstract class ActivityScreenConfigurator(intent: Intent?) : BaseActivityViewConfigurator<AppComponent, ActivityComponent, DefaultActivityScreenModule>(intent) {

    override fun createActivityComponent(parentComponent: AppComponent): ActivityComponent {
        return DaggerActivityComponent.builder()
                .appComponent(parentComponent)
                .defaultActivityModule(DefaultActivityModule(persistentScope))
                .activityNavigationModule(ActivityNavigationModule())
                .build()
    }

    override fun getParentComponent(): AppComponent {
        val app = persistentScope.screenState.activity.application as App
        return app.appComponent
    }

    override fun getActivityScreenModule(): DefaultActivityScreenModule {
        return DefaultActivityScreenModule(persistentScope)
    }
}